src/v/rpc/connection_cache.h

```diff
@@ -57,6 +57,32 @@ class connection_cache final
     /// \brief closes all connections
     ss::future<> stop();
 
+    template<typename Protocol, typename Func>
+    requires requires(Func&& f, Protocol proto) { f(proto); }
+    auto with_node_client_locally(
+      model::node_id node_id,
+      clock_type::time_point connection_timeout,
+      Func&& f) {
+        using ret_t = result_wrap_t<std::invoke_result_t<Func, Protocol>>;
+
+        if (!contains(node_id)) {
+            // No client available
+            return ss::futurize<ret_t>::convert(
```

Comment:
maybe we can leverage coroutines here ?

Reply:
I gave it go, but the compiler couldn't infer the promise type (even when I was explicit about the return type of the function).